zng/recordval.go

```diff
@@ -56,6 +57,16 @@ type Record struct {
 func NewRecord(typ *TypeRecord, raw zcode.Bytes) *Record {
 	return &Record{
 		Type:        typ,
+		Alias:       typ,
+		nonvolatile: true,
+		Raw:         raw,
+	}
+}
+
+func NewRecordFromType(typ Type, raw zcode.Bytes) *Record {
+	return &Record{
+		Type:        AliasedType(typ).(*TypeRecord),
+		Alias:       typ,
 		nonvolatile: true,
 		Raw:         raw,
 	}
```

Comment:
Instead of adding NewRecordFromType et al., maybe just change the `typ *TypeRecord` to `typ Type`?
```
func NewRecord(typ Type, raw zcode.Bytes) *Record {
	return &Record{
		Type:        AliasedType(typ).(*TypeRecord),
		Alias:       typ,
		nonvolatile: true,
		Raw:         raw,
	}
}

Reply:
I did this on purpose to document the different call sites with different semantics for when we refactor in the next step and don't want to miss the variations in use cases.  I was presuming all the NewRecord* functions will be cleaned up and normalized at that point.